### PR TITLE
feat(plasma-web/b2c): Merge Tabs component

### PR DESCRIPTION
**Pull request by Yakutoc**
Плановое объеденение компонентов. 

Упростили и теперь из `plasma-hope` идет **re-export** компонента **Tabs** в 

- b2c
- web

:baby_chick: Download canary assets:

[colors--canary.445.4554428784.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.445.4554428784.xml)  
[color_metro_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.445.4554428784.swift)  
[color_metro_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.445.4554428784.kt)  
[color_metro_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.445.4554428784.ts)  
[color_sberHealth_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.445.4554428784.swift)  
[color_sberHealth_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.445.4554428784.kt)  
[color_sberHealth_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.445.4554428784.ts)  
[color_sbermarket_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.445.4554428784.swift)  
[color_sbermarket_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.445.4554428784.kt)  
[color_sbermarket_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.445.4554428784.ts)  
[color_sberprime_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.445.4554428784.swift)  
[color_sberprime_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.445.4554428784.kt)  
[color_sberprime_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.445.4554428784.ts)  
[color_selgros_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.445.4554428784.swift)  
[color_selgros_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.445.4554428784.kt)  
[color_selgros_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.445.4554428784.ts)  
[color_smbusiness_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.445.4554428784.swift)  
[color_smbusiness_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.445.4554428784.kt)  
[color_smbusiness_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.445.4554428784.ts)  
[PlasmaTokensColor--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.445.4554428784.swift)  
[shadow_sbermarket_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.445.4554428784.ts)  
[typo_mage_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.445.4554428784.swift)  
[typo_mage_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.445.4554428784.kt)  
[typo_mage_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.445.4554428784.ts)  
[typo_plasma_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.445.4554428784.swift)  
[typo_plasma_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.445.4554428784.kt)  
[typo_plasma_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.445.4554428784.ts)  
[typo_ruler_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.445.4554428784.swift)  
[typo_ruler_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.445.4554428784.kt)  
[typo_ruler_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.445.4554428784.ts)  
[typo_sage_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.445.4554428784.swift)  
[typo_sage_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.445.4554428784.kt)  
[typo_sage_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.445.4554428784.ts)  
[typo_sbermarket_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.445.4554428784.swift)  
[typo_sbermarket_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.445.4554428784.kt)  
[typo_sbermarket_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.445.4554428784.ts)  
[typo_soulmate_ios-swift--canary.445.4554428784.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.445.4554428784.swift)  
[typo_soulmate_kotlin--canary.445.4554428784.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.445.4554428784.kt)  
[typo_soulmate_react-native--canary.445.4554428784.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.445.4554428784.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.170.0-canary.445.4554428784.0
  npm install @salutejs/plasma-hope@0.28.0-canary.445.4554428784.0
  npm install @salutejs/plasma-web@1.195.0-canary.445.4554428784.0
  # or 
  yarn add @salutejs/plasma-b2c@1.170.0-canary.445.4554428784.0
  yarn add @salutejs/plasma-hope@0.28.0-canary.445.4554428784.0
  yarn add @salutejs/plasma-web@1.195.0-canary.445.4554428784.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-445/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-445/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-445/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-445/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-445/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-445/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-445/temple-storybook/

**Comment by Yeti-or** [comment]
prop `design` можно из propTable тоже дронуть?

**Comment by Yakutoc** [comment]
> prop `design` можно из propTable тоже дронуть?

Можно сделать exclude  в документации. 

На уровне типов пока не нашел способ как его убрать.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-445/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-445/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-445/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-web/src/components/Tabs/TabItem.tsx [hunk 1]
```diff
@@ -1,43 +1,8 @@
...
-            color: ${link};
-            box-shadow: inset 0 -0.125rem 0 ${link};
-        `}
-
-    &.focus-visible:focus {
-        color: ${linkActive};
-        box-shadow: inset 0 -0.125rem 0 ${linkActive};
-    }
-`;
+export const TabItem = styled(TabItemHope)<TabItemProps>``;
```
Comment (neretin-trike) [inline packages/plasma-web/src/components/Tabs/TabItem.tsx:8]: поскольку TabItem одинаковый для B2C и Web, я считаю, что не нужно держать отдельные файлы для него и дублировать код, можно сделать просто реэкспорт из plasma-hope

File: packages/plasma-b2c/src/components/Tabs/TabItem.tsx [hunk 2]
```diff
@@ -1,53 +1,8 @@
...
-            outlineRadius: 'var(--tab-focus-border-radius)',
-            outlined: true,
-            outlineSize: '0.063rem',
-            outlineOffset: '0.125rem',
-            outlineColor: accent,
-            lowPerformance,
-        })}
-    `}
-`;
+export const TabItem = styled(TabItemHope).attrs({ design: 'b2c' })<TabItemProps>``;
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Tabs/TabItem.tsx:8]: поскольку TabItem одинаковый для B2C и Web, я считаю, что не нужно держать отдельные файлы для него и дублировать код, можно сделать просто реэкспорт из plasma-hope

File: packages/plasma-b2c/src/components/Tabs/Tabs.stories.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { InSpacingDecorator, disableProps } from '@salutejs/plasma-sb-utils';
 
 import { Tabs, TabsProps, TabItem, TabsController } from '.';
 
-const propsToDisable = ['ref', 'theme', 'as', 'forwardedAs'];
+const propsToDisable = ['ref', 'theme', 'as', 'forwardedAs', 'design'];
```
Comment (neretin-trike) [inline packages/plasma-b2c/src/components/Tabs/Tabs.stories.tsx:10]: лучше выдрать это из типа

File: packages/plasma-hope/src/components/Tabs/TabItem.tsx [hunk 4]
```diff
@@ -1,43 +1,21 @@
...
+import { TabItem as TabItemWeb } from './views/web/TabItem';
+import type { Design, TabItemProps } from './types';
+
+const componentMap = {
+    b2c: TabItemB2C,
+    web: TabItemWeb,
+};
+
+export const TabItem: FC<PropsWithChildren<TabItemProps & Design>> = (props) => {
+    const { design = 'web', children, as, ...rest } = props;
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/Tabs/TabItem.tsx:13]: предлагаю не указывать значение по умолчанию, а сделать это поле строго обязательным, для лучшей читаемости

File: packages/plasma-hope/src/components/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
-    --tab-focus-border-size: 0rem;
+const componentMap = {
+    b2c: TabsB2C,
+    web: TabsWeb,
+};
 
-    gap: 0;
-`;
+export const Tabs = forwardRef<HTMLDivElement, PropsWithChildren<TabsProps & Design>>((props, ref) => {
+    const { design = 'web', children, as, ...rest } = props;
```
Comment (neretin-trike) [inline packages/plasma-hope/src/components/Tabs/Tabs.tsx:13]: здесь также считаю, что не нужно указывать значение по умолчанию

File: packages/plasma-web/src/components/Tabs/Tabs.tsx [hunk 6]
```diff
@@ -1,15 +1,8 @@
...
 /**
  * Контейнер вкладок, основной компонент для пользовательской сборки вкладок.
  */
-export const Tabs = styled(BaseTabs)<TabsProps>`
-    --tabs-margin: 1rem;
-    --tab-focus-border-size: 0rem;
-
-    gap: 0;
-`;
+export const Tabs = styled(TabsHope)<TabsProps>``;
```
Comment (neretin-trike) [inline packages/plasma-web/src/components/Tabs/Tabs.tsx:8]: здсь также кажется, что надо добавить `.attrs({ design: 'web' })`

File: website/plasma-web-docs/docs/components/Tabs.mdx [hunk 7]
```diff
@@ -19,12 +19,12 @@ import { PropsTable, Description, StorybookLink } from '@site/src/components';
...
 
 <Description name="Tabs" />
-<PropsTable name="Tabs" />
+<PropsTable name="Tabs" exclude='design'/>
 
 ## TabItem
 
 <Description name="TabItem" />
-<PropsTable name="TabItem" />
+<PropsTable name="TabItem" exclude='design' />
```
Comment (neretin-trike) [inline website/plasma-web-docs/docs/components/Tabs.mdx:27]: лучше это разрешить на уровне типов

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-445/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-445/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-445/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-445/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-445/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-445/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-445/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-445/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**